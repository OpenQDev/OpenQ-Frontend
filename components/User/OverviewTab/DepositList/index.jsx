// Third Party
import React from 'react';

// Custom
import DepositCard from '../../../RefundBounty/DepositCard';

const DepositList = ({ deposits }) => {
  return (
    <div className='px-10 py-10 border-web-gray border-b'>
      <h2 className='font-bold uppercase text-gray-300 text-xl pb-4 px-6'>Deposits</h2>

      {deposits.length > 0 ? (
        <ul className='flex flex-wrap justify-between gap-5 pt-2'>
          {deposits.map((deposit) => (
            <DepositCard key={deposit.id} showLink={true} deposit={deposit} />
          ))}
        </ul>
      ) : (
        <div className='px-6 pt-2'>No Deposits</div>
      )}
    </div>
  );
};

export default DepositList;
