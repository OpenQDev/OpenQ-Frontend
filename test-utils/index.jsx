// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
