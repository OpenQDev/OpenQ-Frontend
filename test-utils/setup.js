import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
