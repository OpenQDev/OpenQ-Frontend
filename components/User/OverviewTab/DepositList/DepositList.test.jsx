/**
 * @vi-environment jsdom
 */
import React from 'react';

import { render, screen } from '../../../../test-utils';
import DepositList from '../../../../components/User/OverviewTab/DepositList';
import ShallowRenderer from 'react-test-renderer/shallow';
import Constants from '../../../../test-utils/constant';
import { cleanup } from '@testing-library/react';

describe('DepositList', () => {
  const deposits = Constants.deposits;

  afterEach(() => {
    cleanup();
  });
  it('should match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<DepositList deposits={deposits} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('should render Bounty heading', async () => {
    // ARRANGE
    render(<DepositList deposits={deposits} />);

    // ASSERT
    expect(screen.getAllByText(/Deposited on: September 7, 2022 at 10:09/)).toHaveLength(3);
    expect(screen.getAllByText(/Refundable on: September 22, 2022 at 10:09/)).toHaveLength(2);
    expect(screen.getByText(/Refunded on: September 7, 2022 at 14:08/)).toBeInTheDocument();
  });
});
