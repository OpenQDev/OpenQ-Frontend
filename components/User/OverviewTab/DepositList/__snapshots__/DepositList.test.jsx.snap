// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DepositList > should match DOM Snapshot 1`] = `
<div
  className="px-10 py-10 border-web-gray border-b"
>
  <h2
    className="font-bold uppercase text-gray-300 text-xl pb-4 px-6"
  >
    Deposits
  </h2>
  <ul
    className="flex flex-wrap justify-between gap-5 pt-2"
  >
    <DepositCard
      deposit={
        {
          "__typename": "Deposit",
          "expiration": "1296000",
          "id": "0xe5551a3fa87d93a0c6c084d572b9e282114befc43dc68f08be6d53d13830e356",
          "receiveTime": "1662545374",
          "refundTime": null,
          "refunded": false,
          "sender": {
            "__typename": "User",
            "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
          },
          "tokenAddress": "0x0000000000000000000000000000000000000000",
          "volume": "1000000000000000000",
        }
      }
      showLink={true}
    />
    <DepositCard
      deposit={
        {
          "__typename": "Deposit",
          "expiration": "30",
          "id": "0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a",
          "receiveTime": "1662545373",
          "refundTime": "1662559726",
          "refunded": true,
          "sender": {
            "__typename": "User",
            "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
          },
          "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
          "volume": "2000000000000000000",
        }
      }
      showLink={true}
    />
    <DepositCard
      deposit={
        {
          "__typename": "Deposit",
          "expiration": "1296000",
          "id": "0x7db2691e573b9c19e90f391cd3eda9ce9246666a1502f2bf87b9d47d13679bc0",
          "receiveTime": "1662545372",
          "refundTime": null,
          "refunded": false,
          "sender": {
            "__typename": "User",
            "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
          },
          "tokenAddress": "0x5fbdb2315678afecb367f032d93f642f64180aa3",
          "volume": "1000000000000000000",
        }
      }
      showLink={true}
    />
  </ul>
</div>
`;
